front/lib/api/files/upload.ts

```diff
@@ -291,121 +280,91 @@ type ProcessingFunction = (
-    folder_table: notSupportedError,
-    conversation: extractTextFromFileAndUpload,
-    avatar: notSupportedError,
-    tool_output: notSupportedError,
-  },
-  "application/pdf": {
-    folder_document: extractTextFromFileAndUpload,
-    folder_table: notSupportedError,
-    conversation: extractTextFromFileAndUpload,
-    avatar: notSupportedError,
-    tool_output: notSupportedError,
-  },
-  "image/jpeg": {
-    conversation: resizeAndUploadToFileStorage,
-    folder_document: notSupportedError,
-    folder_table: notSupportedError,
-    avatar: uploadToPublicBucket,
-    tool_output: storeRawText,
-  },
-  "image/png": {
-    conversation: resizeAndUploadToFileStorage,
-    folder_document: notSupportedError,
-    folder_table: notSupportedError,
-    avatar: uploadToPublicBucket,
-    tool_output: notSupportedError,
-  },
-  "text/comma-separated-values": {
-    conversation: extractContentAndSchemaFromDelimitedTextFiles,
-    folder_document: storeRawText,
-    folder_table: extractContentAndSchemaFromDelimitedTextFiles,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/csv": {
-    conversation: extractContentAndSchemaFromDelimitedTextFiles,
-    folder_document: storeRawText,
-    folder_table: extractContentAndSchemaFromDelimitedTextFiles,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/markdown": {
-    conversation: storeRawText,
-    folder_document: storeRawText,
-    folder_table: notSupportedError,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/plain": {
-    conversation: storeRawText,
-    folder_document: storeRawText,
-    folder_table: notSupportedError,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/tab-separated-values": {
-    conversation: extractContentAndSchemaFromDelimitedTextFiles,
-    folder_document: storeRawText,
-    folder_table: extractContentAndSchemaFromDelimitedTextFiles,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/tsv": {
-    conversation: extractContentAndSchemaFromDelimitedTextFiles,
-    folder_document: storeRawText,
-    folder_table: extractContentAndSchemaFromDelimitedTextFiles,
-    avatar: notSupportedError,
-    tool_output: storeRawText,
-  },
-  "text/vnd.dust.attachment.slack.thread": {
-    conversation: storeRawText,
-    folder_document: notSupportedError,
-    folder_table: notSupportedError,
-    avatar: notSupportedError,
-    tool_output: notSupportedError,
-  },
+export const isUploadSupported = (arg: {
+  contentType: SupportedFileContentType;
+  useCase: FileUseCase;
+}): boolean => {
+  const processing = getProcessingFunction(arg);
+  return !!processing;
 };
 
 const maybeApplyProcessing: ProcessingFunction = async (
   auth: Authenticator,
   file: FileResource
 ) => {
-  const contentTypeProcessing = processingPerContentType[file.contentType];
-  if (!contentTypeProcessing) {
```

Comment:
this could not happen anyway as the mapping table forced us to list all contentType.